### ISSUE TITLE
Callable is assumed to be a closure

### ISSUE THREAD

**Issue by marcguyer** [tuupola/cors-middleware#26]
In a few locations, where `is_callable()` or the `callable` type hint is used, the value is assumed to be a closure and fails when the callable uses, e.g., `['class', 'method']` notation. Here are, I think, all of the instances of this:

https://github.com/tuupola/cors-middleware/blob/bd36c15e1abc63f300a004fa7da850ba845b92a5/src/CorsMiddleware.php#L154
https://github.com/tuupola/cors-middleware/blob/bd36c15e1abc63f300a004fa7da850ba845b92a5/src/CorsMiddleware.php#L193
https://github.com/tuupola/cors-middleware/blob/bd36c15e1abc63f300a004fa7da850ba845b92a5/src/CorsMiddleware.php#L261
https://github.com/tuupola/cors-middleware/blob/bd36c15e1abc63f300a004fa7da850ba845b92a5/src/CorsMiddleware.php#L243

When using `is_callable()` you can [use the third arg](http://php.net/manual/en/function.is-callable.php) to get the callable name for the purposes of calling it, which also will properly call a callable that isn't a closure. So, for example, something like this: 

```php
if (is_callable($this->options["methods"], false, $callableName)) {
    $methods = (array) $callableName($request, $response);
} else {
    $methods = $this->options["methods"];
}
```

For the type hint issue, the type hint can stay the same, then convert (new in 7.1!) to a closure if it isn't a closure. So, for example:

```php
private function error(callable $error): void
{
    if (! $error instanceof Closure) {
        $error = Closure::fromCallable($error);
    }
    $this->options['error'] = $error->bindTo($this);
}
```

Of course there's a workaround: simply convert to a closure at implementation but that just ain't right.

**Comment by marcguyer** [comment]
Actually it turns out that my suggested workaround doesn't work. When using `Closure::fromCallable`, you get a "fake" closure. See here: https://stackoverflow.com/a/47168065/1774759

So, the workaround is to wrap your callable in a "real" closure at implementation.

**Comment by Danack** [comment]
There is code in the Auryn DIC that normalises all callables, to be, er.....callable, which might be relevant and useful to steal ^H^H^H be inspired by.

https://github.com/rdlowrey/auryn/blob/master/lib/Executable.php

**Comment by tuupola** [comment]
Somewhat related: https://github.com/tuupola/slim-jwt-auth/issues/143. I will look into this. 

@marcguyer which workaround you ended up currently using?

**Comment by marcguyer** [comment]
I wrapped my call in a closure before providing it in the options array.

**Comment by tuupola** [comment]
This should be fixed in `0.9.2`.

**Comment by marcguyer** [comment]
Confirmed this is working as expected in my implementation. Thanks for the improvement!
